### PR TITLE
chore(Menu): convert Menu.stories.js to .tsx

### PR DESCRIPTION
**Pull request by itaim18**
- [x] I have read the [Contribution Guide](../packages/core/CONTRIBUTING.md) for this project.
This solve issue #2206

Reluctently, I was also having to convert `...\core\src\components\GridKeyboardNavigationContext\__stories__\useGridKeyboardNavigationContext.stories.helpers.tsx` but seeing affected files it didn't seem to have a huge change on the expected beahvior of them.

Thanks for opening up to other conributors, I am VIBEing this :)
putting-on-my-sunglasses-ken

### CODE REVIEW

File: packages/core/src/components/Menu/Menu/__stories__/Menu.stories.tsx [hunk 1]
```diff
@@ -5,13 +5,23 @@ import { DialogContentContainer, Menu, MenuDivider, MenuGridItem, MenuItem, Menu
...
   component: Menu
 };
 
-const menuTemplate = args => (
+const menuTemplate = (
+  args: VibeComponent<MenuProps> & {
+    isMenu?: boolean;
+    supportFocusOnMount?: boolean;
+    sizes?: typeof SIZES;
+  }
```
Comment (talkor) [inline packages/core/src/components/Menu/Menu/__stories__/Menu.stories.tsx:23]: What if you just use `MenuProps` ? Will it still require these static props?

**Review by talkor — APPROVED** [review]
Awesome! Thank you @itaim18! 🙏🏼

File: packages/core/src/components/GridKeyboardNavigationContext/__stories__/useGridKeyboardNavigationContext.stories.helpers.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
+import { range } from "lodash-es";
+import React, { forwardRef, useMemo, useCallback, useRef, MutableRefObject, RefObject, ForwardedRef } from "react";
```
[the remark this answers is not in the document]
Comment (itaim18) [reply packages/core/src/components/GridKeyboardNavigationContext/__stories__/useGridKeyboardNavigationContext.stories.helpers.tsx:2]: On it

File: packages/core/src/components/Menu/Menu/__stories__/Menu.stories.tsx (hunk 1 above)
Comment (itaim18) [reply packages/core/src/components/Menu/Menu/__stories__/Menu.stories.tsx:23]: It would not :)

**Comment by talkor** [comment]
@itaim18 lint failed for the new interfaces, can you please check? 
You can run `yarn lint` to verify there are no errors 🙏🏼 
Screenshot 2024-07-24 at 16 24 00

File: packages/core/src/components/GridKeyboardNavigationContext/__stories__/useGridKeyboardNavigationContext.stories.helpers.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    );
+  }
+);
+
+DummyNavigableGrid.defaultProps = {
+  itemPrefix: "",
+  disabled: false,
+  disabledIndexes: [],
+  withoutBorder: false
+};
```
Comment (talkor) [inline packages/core/src/components/GridKeyboardNavigationContext/__stories__/useGridKeyboardNavigationContext.stories.helpers.tsx:70]: @itaim18 Let's use defaults in the component's signature :) Although it should work perfectly fine, in this project we define defaults in the signature

Comment (itaim18) [reply packages/core/src/components/GridKeyboardNavigationContext/__stories__/useGridKeyboardNavigationContext.stories.helpers.tsx]: 😅

**Comment by talkor** [comment]
Thank you @itaim18 !!

**Comment by itaim18** [comment]
😃

**Pull request merged**